container/go/pkg/oci/write_test.go

```diff
@@ -0,0 +1,188 @@
+	// content is the oci layout content where the key is the file path and value is sha256 sum
+	content map[string]v1.Hash
+}{
+	{
+		"alpine_linux",
+		"testdata/test_write1/small_linux.tar",
+		"sha256/9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c",
+		map[string]v1.Hash{
+			"blobs/sha256/9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c": v1.Hash{Algorithm: "sha256", Hex: "9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c"},
+			"index.json": v1.Hash{Algorithm: "sha256", Hex: "ce6ed2e6817add8a5a0e94f4c2ac00360c7ca2cc98c4972b86530f14de622508"},
+			"oci-layout": v1.Hash{Algorithm: "sha256", Hex: "b66dbb27a73334db6ac9c030475837bd7f4472d835c72b2360534b203edce6cb"},
+		},
+	},
+}
+
+// TestWrite checks to ensure that the v1.Image written is in OCI format by validating image output format and content
+func TestWrite(t *testing.T) {
+	for _, wt := range testCases {
+		t.Run(wt.name, func(t *testing.T) {
+			tmp, err := ioutil.TempDir("", "write-oci-index-test")
+			if err != nil {
+				t.Errorf("Write(%s): %v", os.TempDir(), err)
+			}
+
+			defer os.RemoveAll(tmp)
+
+			img, err := tarball.ImageFromPath(wt.dataPath, nil)
+			if err != nil {
+				t.Errorf("error loading image from testdata: %v", err)
+			}
+
+			if err := assertImageIndexValid(tmp, img); err != nil {
+				t.Errorf("image written is not a valid index: %v", err)
+			}
+
+			if err = assertOCIFormat(tmp, img, wt.content, wt.shaPath); err != nil {
+				t.Errorf("image is not in oci format: %v", err)
+			}
+		})
+	}
+}
+
+func assertImageIndexValid(tmp string, img v1.Image) error {
+	if err := Write(img, tmp); err != nil {
+		return fmt.Errorf("error writing image to temp path %s", tmp)
+	}
+
+	written, err := layout.ImageIndexFromPath(tmp)
+	if err != nil {
+		return fmt.Errorf("failed to write image to temp path %s", tmp)
+	}
+
+	if err := validate.Index(written); err != nil {
+		return fmt.Errorf("validate.Index() = %v", err)
+	}
+
+	return nil
+}
+
+func assertOCIFormat(tmp string, img v1.Image, content map[string]v1.Hash, shaPth string) error {
+	if err := Write(img, "/usr/local/google/home/xwinxu/Desktop/temp/temp"); err != nil {
+		return fmt.Errorf("error writing image to temp path %s", tmp)
+	}
+
+	// validate that the contents of blobs/ were written in the following format:
+	// Top level:
+	//     oci-layout, index.json, blobs/
+	// Under blobs/, for each image:
+	// 	   one file for each layer, config blob, and manifest blob
+	blobs := path.Join(tmp, blob)
+	file := path.Join(blobs, "sha256", "9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c")
+	if err := validateDir(blobs, content["blobs/sha256/9a96f3888ebad00d46bca04ccb591e70d091624835998668af551f48512d9b5c"].Hex, tmp, file); err != nil {
+		return fmt.Errorf("failed to validate blobs/ directory: %v", err)
+	}
+
+	// validate the index.json file
+	if err := validateFile(path.Join(tmp, index), index, content["index.json"].Hex, tmp); err != nil {
```

Comment:
Try to put all these checks into a loop that iterates over the keys of your content map. Otherwise, we are duplicate and hardcode the keys again here.